cli/src/diff_util.rs

```diff
@@ -237,6 +272,46 @@ impl<'a> DiffRenderer<'a> {
         }
     }
 
+    /// Generates diff of the given `commit` compared to its parents.
+    pub fn show_patch(
+        &self,
+        ui: &Ui,
+        formatter: &mut dyn Formatter,
+        commit: &Commit,
+        matcher: &dyn Matcher,
+    ) -> Result<(), DiffRenderError> {
+        let from_tree = commit.parent_tree(self.repo)?;
+        let to_tree = commit.tree()?;
+        self.show_diff_impl(
+            ui,
+            formatter,
+            &from_tree,
+            &to_tree,
+            matcher,
+            &Default::default(),
+        )
+    }
+
+    /// Generates diff between two commits.
+    pub fn show_diff_commits(
+        &self,
+        ui: &Ui, // TODO: remove Ui dependency if possible
+        formatter: &mut dyn Formatter,
+        from: &Commit,
+        to: &Commit,
```

Comment:
Perhaps, `from` has to be `&[Commit]`?

It might make sense to special case `from.len() == 1`, but we'll eventually need to handle copy info in a merge commit properly.

Reply:
I think you are correct, but I have found it very helpful for understanding how the parts fit together to aggressively simplify.  I believe that after I do a bit more down this path, I will be able to make the CopyRecordsMap thing handle multiple parents before long (at which point the concept of a conflict will reappear but in a more restricted context (I think)

Reply:
we now handle this better, but there is still a TODO for conflicts on the merge